sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredential.java

```diff
@@ -75,13 +75,23 @@ public class AzurePipelinesCredential implements TokenCredential {
                     HttpRequest request = new HttpRequest(HttpMethod.POST, url);
                     request.setHeader(HttpHeaderName.AUTHORIZATION, "Bearer " + systemAccessToken);
                     request.setHeader(HttpHeaderName.CONTENT_TYPE, "application/json");
+                    request.setHeader(HttpHeaderName.fromString("X-TFS-FedAuthRedirect"), "Suppress");
```

Comment:
nit: consider adding a source comment to answer the question for future maintainers, what does this header do (which @KarishmaGhiya [had asked initially](https://github.com/Azure/azure-sdk-for-cpp/pull/6019#discussion_r1769359658)):
> // Prevents the service from responding with a redirect HTTP status code (useful for automation).